hedera-node/src/main/java/com/hedera/services/ledger/HederaLedger.java

```diff
@@ -229,19 +229,19 @@ public TransferList netTransfersInTxn() {
 		for (int i = 0; i < numTouches; i++) {
 			var token = tokensTouched[i];
 			if (i == 0 || !token.equals(tokensTouched[i - 1])) {
-				final var fungibleTransfersHere = netTokenTransfers.get(token);
-				if (fungibleTransfersHere != null) {
-					purgeZeroAdjustments(fungibleTransfersHere);
+				final var uniqueTransfersHere = uniqueTokenTransfers.get(token);
+				if (uniqueTransfersHere != null) {
 					all.add(TokenTransferList.newBuilder()
 							.setToken(token)
-							.addAllTransfers(fungibleTransfersHere.getAccountAmountsList())
+							.addAllNftTransfers(uniqueTransfersHere.getNftTransfersList())
 							.build());
 				} else {
-					var uniqueTransfersHere = uniqueTokenTransfers.get(token);
-					if (uniqueTransfersHere != null) {
+					final var fungibleTransfersHere = netTokenTransfers.get(token);
+					if (fungibleTransfersHere != null) {
+						purgeZeroAdjustments(fungibleTransfersHere);
 						all.add(TokenTransferList.newBuilder()
 								.setToken(token)
-								.addAllNftTransfers(uniqueTransfersHere.getNftTransfersList())
+								.addAllTransfers(fungibleTransfersHere.getAccountAmountsList())
 								.build());
```

Comment:
I am not sure why we need to change this logic. Maybe a mistake from the merge?

Reply:
L236 is for nfts, L244 is for fungibles. This is the correct code since we need to calculate first nft and then fungibles